### DISCUSSION TITLE
Request for comments: Auto-archiving old orders

### DISCUSSION THREAD

**Discussion by jamesckemp** [woocommerce/woocommerce#51478] [General]
Hi all 👋🏻

One of our most popular feature requests for WooCommerce is the ability to [automatically archive old orders](https://woocommerce.com/feature-request/auto-archive-old-orders/).

This request originated before the introduction of HPOS (High-Performance Order Storage), when orders were stored as posts and post meta, which had performance implications. Now that HPOS provides dedicated tables for orders and order meta, we'd like your input:

1. Is automatic order archiving still a highly sought-after feature in the HPOS era?
2. Should we instead focus on encouraging users to migrate to HPOS if database performance is their primary concern?
3. Would adding a new "Archived" order status still be valuable to help merchants keep their orders list clutter-free and more actionable?

We greatly value your thoughts on this feature request and how we should approach it. Your feedback will help shape our decision-making process.

Thank you for your input!

**Comment by senadir** [comment]
How would this work with flows like ones that check customer history and how much they spent/orders placed? I use those flows in my store. Admittedly, I only have 3k orders, so I haven't seen any bottlenecks yet (except in search).

**Reply by jamesckemp** [reply]
It's just an order status, so all the data would still exist.

**Reply by masteradhoc** [reply]
I guess HPOS has solved and will continue to solve the performance issues that made this feature request rise. I still think to keep the order lists more clean a status to archive orders would help. In my opinion i see the following useful:
- Setting to define after which time a completed order should be archived (similar to gdpr settings)
- Move completed orders after x days/weeks/years automatically to the archive status

On Order Overview where "All Status" Orders show the archive status shouldnt be part.
